src/config/load_config.c

```diff
@@ -18,70 +18,98 @@
 ///     The map is loaded in the linked list llmap.
 int	load_config(char *file, t_cub3d *world)
 {
-	// Verify ".cub" extension
+	if (ft_strncmp(file + ft_strlen(file) - 4, ".cub", 4) != 0)
+		exit_error(ECUB);
 	parse_config(file, world);
-	translate_map(world);
-	if (valid_perimeter(world->map) == EXIT_FAILURE)
-		exit_error(EPERIMETER);
+	translate_map(world->llmap, world);
+	valid_perimeter(world);
 	return (EXIT_SUCCESS);
 }
 
 // unknown key, double keys, an invalid path,
 // invalid color, invalid map, invalid texture
-int	parse_element(char **element, t_cub3d *word)
+void	parse_element(char **element, t_cub3d *word)
 {
-	// Check if the first element is:
-	// NO, SO, WE, EA, F, C
-	// Check if the texture is valid
-	// or Check if the color is valid [0-255,0-255,0-255]
-	// word->graphics.ceiling_color = ft_rgb_to_int(r, g, b);
-	// On error, call ft_free_split(element) and exit_error();
-	// On success, save the element in the structure
+	char	*id;
+
+	if (element)
+	{
+		if (element[2])
+			exit_error(EINFO);
+		id = element[0];
+		if (ft_strcmp(id, "NO") == 0 || ft_strcmp(id, "SO") == 0 \
+			|| ft_strcmp(id, "WE") == 0 || ft_strcmp(id, "EA") == 0)
+			check_nsew(element, word);
+		else if (ft_strcmp(id, "C") == 0 || ft_strcmp(id, "F") == 0)
+			check_cf(element, word);
+		else
+		{
+			ft_free_split(&element);
+			exit_error(EUNKNOWN);
+		}
+	}
 }
 
-int	parse_map(char *line, t_cub3d *world)
+void	parse_map(char *line, t_cub3d *world)
 {
+	int	i;
 	int	size;
 
 	size = ft_strlen(line);
-	if (size == 0)
-		exit_error(EELMAP);
 	if (size > world->map_w)
 		world->map_w = size;
-	// Check if the line is a valid map line
-	// On error, exit_error();
-	// On success, save the line in the map
+	i = 0;
+	while (line[i])
+	{
+		if (line[i] != '0' && line[i] != '1' && line[i] != 'N' && line[i] != 'S'\
+			&& line[i] != 'E' && line[i] != 'W' && line[i] != ' ')
+			exit_error(EIEMAP);
+		if (line[i] == 'N' || line[i] == 'S' || \
+			line[i] == 'E' || line[i] == 'W')
```

Comment:
The backslash `\` at the end of a line allows to bypass the norme.

I propose this alternative, that goes more according to the norme and maintain readability and adherence to coding standards.

```C
if (line[i] == 'N' || line[i] == 'S'
	|| line[i] == 'E' || line[i] == 'W')
```